### DISCUSSION TITLE
NPM DNS challenge cloudflare

### DISCUSSION THREAD

**Discussion by ilgio** [tteck/Proxmox#3859] [General] [answered]
I followed the guide to allow the certbot to cloudflare with the creation of API but it gives me an error

certbot: error: unrecognized arguments: --dns-cloudflare-credentials /etc/letsencrypt/credentials/credentials-9

<img width="810" alt="Screenshot 2024-10-07 alle 16 17 41" src="https://github.com/user-attachments/assets/20050652-859e-4072-be2d-2cf0a65cd4d9">
<img width="1311" alt="Screenshot 2024-10-07 alle 16 17 16" src="https://github.com/user-attachments/assets/9bbbe647-d207-4d08-89b9-e23c6858e6d4">
<img width="596" alt="Screenshot 2024-10-07 alle 16 16 56" src="https://github.com/user-attachments/assets/c2e436c5-97bd-4bbc-9379-ed9fb0428c95">

**Comment by ilgio** [comment, marked as answer]
I resolve it
[https://techoverflow.net/2022/08/19/how-to-fix-certbot-error-unrecognized-arguments-dns-cloudflare-credentials/](url)

`sudo pip3 install --upgrade certbot-dns-cloudflare
`

this is a bug... it should be put in NPM LXC package

**Reply by Mellowlynx** [reply]
As it stated on the website: Since there are hundreds of Certbot instances, it's necessary to install the specific Certbot of your preference.

![image](https://github.com/user-attachments/assets/35f5dfa8-08df-452d-a58e-dba957451eb0)

**Reply by ilgio** [reply]
Are there a example?

**Reply by nihkel** [reply]
> I resolve it [https://techoverflow.net/2022/08/19/how-to-fix-certbot-error-unrecognized-arguments-dns-cloudflare-credentials/](url)
> 
> `sudo pip3 install --upgrade certbot-dns-cloudflare `
> 
> this is a bug... it should be put in NPM LXC package

i've been fighting this for hours. You are a hero in my book. Thanks
